cmd/gateway-handlers.go

```diff
@@ -246,6 +246,13 @@ func (api gatewayAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Re
 		}
 	}
 
+	// Make sure we hex encode md5sum here.
+	metadata["etag"] = hex.EncodeToString(md5Bytes)
```

Comment:
are you using this anywhere? it does not look like.

Reply:
@stack72 @cheapRoc can you fix these as well?